app/controllers/scans_controller.rb

```diff
@@ -33,17 +33,30 @@ def current_request
 
   protected
 
+  def illiad_page
+    @illiad_page = { 'Action': '10', 'Form': '30', 'rft.genre': 'scananddeliverArticle' }
+  end
+
+  def illiad_item_data(scan)
+    @illiad_item_data = { 'rft.location': scan.origin,
+                          'rft.callnum': scan.holdings.first.callnumber,
+                          'rft.item': scan.holdings.first.barcode }
+  end
+
+  def illiad_citation(scan)
+    @illiad_params = { 'rft.jtitle': scan.item_title,
+                       'rft.au': scan.data[:authors],
+                       'rft.pages': scan.data[:page_range],
+                       'rft.atitle': scan.data[:section_title] }
+  end
+
+  def illiad_callback(scan)
+    @illiad_callback = { 'scan_referrer': successful_scan_url(scan) }
+  end
+
   def illiad_query(scan)
```

Comment:
We didn’t talk anything through. He just mentioned it at the standup, so I started giving it a try, but it looks like master has now changed, and my commits will now cause many conflicts, so back to square 1.

> On Jun 12, 2015, at 2:48 PM, Jessie Keck <notifications@github.com> wrote:
> 
> In app/controllers/scans_controller.rb <https://github.com/sul-dlss/sul-requests/pull/179#discussion_r32358231>:
> 
> > +    @illiad_item_data = { 'rft.location': scan.origin,
> > +                          'rft.callnum': scan.holdings.first.callnumber,
> > +                          'rft.item': scan.holdings.first.barcode }
> > +  end
> > +
> > +  def illiad_citation(scan)
> > +    @illiad_params = { 'rft.jtitle': scan.item_title,
> > +                       'rft.au': scan.data[:authors],
> > +                       'rft.pages': scan.data[:page_range],
> > +                       'rft.atitle': scan.data[:section_title] }
> > +  end
> > +
> > +  def illiad_callback(scan)
> > +    @illiad_callback = { 'scan_referrer': successful_scan_url(scan) }
> > +  end
> > +
> >    def illiad_query(scan)
>  I wasn't aware you guys talked it through.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/sul-dlss/sul-requests/pull/179/files#r32358231>.
>